<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en-US" lang="en-US">
<head>
  <link href="https://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta name="referrer" content="never" />
  <meta name="generator" content="Hugo 0.104.3" />

  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">

  
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.css">
  <script src="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.js"></script>
  
  <script src="https://cdn.jsdelivr.net/npm/meting@2/dist/Meting.min.js"></script>  

  <title>Natural product function &middot; C&#39;s Blog</title>
  <meta name="description" content="" />

  
  <link type="text/css" rel="stylesheet" href="Chocomilk-Leo.github.io/css/print.css" media="print">
  <link type="text/css" rel="stylesheet" href="Chocomilk-Leo.github.io/css/poole.css">
  <link type="text/css" rel="stylesheet" href="Chocomilk-Leo.github.io/css/syntax.css">
  <link type="text/css" rel="stylesheet" href="Chocomilk-Leo.github.io/css/hyde.css">
    <link rel="stylesheet" href="https://fonts.loli.net/css?family=Abril+Fatface|PT+Sans:400,400i,700">


  
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/apple-touch-icon-144-precomposed.png">
  <link rel="shortcut icon" href="/favicon.png">

  
  
</head>

  <body class=" ">
  <aside class="sidebar">
  <div class="container sidebar-sticky">
    <div class="sidebar-about">
      <a href="Chocomilk-Leo.github.io/"></a>
      <a class="site-title" href="Chocomilk-Leo.github.io/">
        <div class="main-title"><h1>Itten's</h1></div>
        <div class="main-title"><h1>Blog</h1></div>
      </a>
      <p class="lead">
       Finding Enthusiasm. 
      </p>
    </div>

    <nav>
      <ul class="sidebar-nav">
        <li><a href="Chocomilk-Leo.github.io/">Home</a> </li>
        <li><a href="Chocomilk-Leo.github.io/posts/"> Archives </a></li><li><a href="Chocomilk-Leo.github.io/photos/"> Photos </a></li><li><a href="Chocomilk-Leo.github.io/tags/"> Tags </a></li>
      </ul>
    </nav>

    <p>&copy; 2022. All rights reserved. </p>
  </div>
</aside>

    <main class="content container">
    <div class="post">
  <h1 class="content-title">Natural product function</h1>
  <div class="post-meta post-date"><time  style="display: inline-block" datetime=2022-10-12T06:28:58Z class="post-meta post-date">2022-10-12</time>

   <div class="post-tags" style="display: inline-block; margin-left: 10px"><a style="margin-left: 5px" href="/tags/[]">#[]</a>
   </div>

  </div>
  
  <div class="post-content" style="margin-top: 20px">
    <p>Natural-product-function
è®ºæ–‡
A Machine Learning Bioinformatics Method to Predict Biological Activity from Biosynthetic Gene Clusters | Journal of Chemical Information and Modeling (acs.org)
ç›¸å…³ç½‘é¡µ
allie-walker/Natural-product-function: scripts for predicting natural product activity from biosynthetic gene cluster sequences (github.com)
æµ‹è¯•ç”¨ä¾‹
MK613929
æ¥æºè®ºæ–‡ï¼šDiscovery and engineering of an endophytic Pseudomonas strain from Taxus chinensis for efficient production of zeaxanthin diglucoside | Journal of Biological Engineering | Full Text (biomedcentral.com)</p>
<p>ä½¿ç”¨æŒ‡å—</p>
<p>inputï¼šFASTA</p>
<p>å…ˆçœ‹ allie-walker/Natural-product-function: scripts for predicting natural product activity from biosynthetic gene cluster sequences (github.com) çš„READMEéƒ¨åˆ†ï¼š</p>
<p>Natural_product_BGC_activity_prediction.ipnyb is an python notebook containing code to optimize, train, and use machine learning models to predict natural product activity from biosynthetic gene clusters. We have also provided a command line version of the script in the cluster_function_prediction.py. The activity prediction program requires ouptut files from antiSMASH and Resistance Gene Identifier (RGI) (the txt file not the JSON). We used antiSMASH version 4.0 and RGI version 3.2.1 with database version 1.1.9, to produce the results described in our manuscript. Since our initial development of the tool newer versions of antiSMASH and RGI were released, we have added support for antiSMASH 5 and RGI 5 to the command line version of our tool and are planning to add support to the python notebook version. Initial analysis showed no significant change in classifier accuracy when updating to the newer versions of antiSMASH and RGI.
Code is developed to run in python 3 (3.8.3). Packages required, version used for publication in parentheses:
Biopython (1.77)
Numpy (1.18.5)
Scikit learn (0.23.1)
Matplotlib (3.2.2)
To generate the proper files, antiSMASH 4 should be run with the command:
antismash &ndash;verbose &ndash;outputfolder OUTPUT_FOLDER &ndash;statusfile status.txt &ndash;full-hmmer &ndash;borderpredict INPUT_FILE
and RGI 3 shoud be run with the command:
rgi -i INPUT_FASTA_FILE -o OUTPUT_FILE &ndash;loose_criteria YES &ndash;verbose ON
Instructions for python notebook: Throughout the cells the classification variable determines which binary classification problem predictions are made for. Options are &ldquo;antibacterial&rdquo;, &ldquo;antieuk&rdquo; (which predicts antifungal, antitumor, or cytotoxic activity), &ldquo;antifungal&rdquo;, &ldquo;cytotoxic_antitumor&rdquo;, &ldquo;antigramneg&rdquo;, or &ldquo;antigrampos.
Cells 1-4 contain methods for processing the training data. These only need to be run if the training set data have been updated.
Cell 5 contains methods used by cells 6-7 and should be run before running those cells.
Cell 6 determines optimal parameters for the different classifiers.
Cell 7 performs cross validation and evaluates classifers by various metrics.
Cell 8 uses the logisitic regression coefficients to analyze which features are most important for the prediction.
Cells 9 and 10 are the only cells that need to be run to make predictions for novel BGCs.
Cell 9 extracts the features from the antiSMASH and rgi output files for the BGC. antiSMASH output genbank files should be saved in the BGCs_for_prediction/antiSMASH folder and rgi output text files should be saved in the BGCs_for_prediction/CARD folder. antiSMASH and RGI files must have matching names so the program knows that they are for the same cluster (e.g. if the antiSMASH genbank file is called cluster1.gbk, the corresponding RGI file must be callsed cluster1.txt).
Predictions can be run either with or without sequence similarity network (sub-PFAM domain) features. Running cell 9 with SSN domains makes the program significantly slower as genes from the cluster need to be blasted against all training set SSN proteins to determine which, if any, sub-PFAM domain the gene belongs to. Running it without SSN features does not significantly decrease the accuracy of the predictions. To run with out SSN features set include_SSN = False in both cell 9 and 10. To run with SSN features set include_SSN = True in both cell 9 and 10 and set the blast_exe variable in cell 9 equal to the path to the blastp executable on your computer.
Cell 10 runs the predictions on the features extracted by cell 9. The classification variable determines which binary prediction will be made, options are &ldquo;antibacterial&rdquo;, &ldquo;antieuk&rdquo; (which predicts antifungal, antitumor, or cytotoxic activity), &ldquo;antifungal&rdquo;, &ldquo;cytotoxic_antitumor&rdquo;, &ldquo;antigramneg&rdquo;, or &ldquo;antigrampos.
Instructions for command line tool: requirements.txt contains all required packages for the command line tool.
To run the command line tool enter:
python cluster_function_prediction.py &ldquo;filename of antismash cluster gbk file&rdquo; &ldquo;filename of rgi txt file&rdquo;
optional arguments include:
&ndash;output - directory where predictions should be saved, current directory by default
&ndash;seed - random seed for predictions
&ndash;no_SSN - use this flag to run without SSN features, makes predictions faster with minimal accuracy cost
&ndash;blastp_path - path to blastp executable, by default assumes blastp is already in PATH variable
&ndash;write_features - directory to write features to, if not defined features will not be saved
&ndash;antismash_version antiSMASH version used to generate antiSMASH input file, support for 4 or 5, default 5
&ndash;rgi_version RGI version to generate RGI input file, support for 3 or 5, default 5
We have also supplied a Dockerfile that can be used to create a Docker image to run the command line tool. To create the docker image use the command: docker build -t function_prediction_docker_img .
To run the docker use the command: docker run &ndash;volume EXACT_PATH_TO_DIRECTORY_WITH_INPUTS_AND_OUTPUTS:/src/input function_prediction_docker_image ./input/ANTISMASH_FILENAME ./input/RGI_FILENAME &ndash;output ./input/OUTPUT_FOLDERNAME additional arguments from the command line tool can also be added to this command</p>
<p>çœ‹å®ŒçŸ¥é“éœ€è¦å…ˆå‡†å¤‡antiSMASHå’Œrgiçš„è¿è¡Œç»“æœï¼Œç„¶åä¸¤è€…çš„ç»“æœä¸€èµ·ä½œä¸ºnatural-product-functionçš„è¾“å…¥</p>
<p>To generate the proper files, antiSMASH 4 should be run with the command:
antismash &ndash;verbose &ndash;outputfolder OUTPUT_FOLDER &ndash;statusfile status.txt &ndash;full-hmmer &ndash;borderpredict INPUT_FILE
and RGI 3 shoud be run with the command:
rgi -i INPUT_FASTA_FILE -o OUTPUT_FILE &ndash;loose_criteria YES &ndash;verbose ON</p>
<p>ä½†æ•™ç¨‹é‡Œçš„å‘½ä»¤å·²ç»è¿‡æ—¶
é‡æ–°è°ƒæ•´å‘½ä»¤ï¼Œæ­¥éª¤å¦‚ä¸‹ï¼š</p>
<ol>
<li>å‡†å¤‡ MK613929.fasta æ–‡ä»¶</li>
<li>è¿›å…¥condaçš„antiSMASHç¯å¢ƒ
conda activate bgc</li>
</ol>
<p>ç¯å¢ƒé…ç½®çš„æˆªå›¾å‚è€ƒå„è½¯ä»¶ç‰ˆæœ¬
3. è¿è¡Œantismash</p>
<p>å‘½ä»¤ï¼š
antismash MK613929.fasta &ndash;genefinding-tool prodigal</p>
<p>ç»“æœï¼š
ç”Ÿæˆäº†ä¸€ä¸ªæ–°ç›®å½•ï¼Œå†…å®¹å¦‚ä¸‹</p>
<ol start="4">
<li>è¿›å…¥condaçš„rgiç¯å¢ƒ</li>
</ol>
<p>ç¯å¢ƒé…ç½®çš„æˆªå›¾å‚è€ƒå„è½¯ä»¶ç‰ˆæœ¬</p>
<ol start="5">
<li>è¿è¡Œrgi</li>
</ol>
<p>å‘½ä»¤ï¼š
rgi main -i MK613929.fasta -o rgi_MK613929</p>
<p>ç»“æœï¼š
åœ¨åŸç›®å½•ä¸‹ç”Ÿæˆäº†ç»“æœ</p>
<ol start="6">
<li>è¿è¡Œnatural-product-function</li>
</ol>
<p>å‘½ä»¤ï¼š
python3 cluster_function_prediction.py /hdata/tmp/seq/anotherANTISMASH/MK613929/MK613929.gbk /hdata/tmp/seq/anotherRGI/rgi_MK613929.txt</p>
<p>ç»“æœï¼š</p>
<p>å¼€è’è®°å½•
*å¡åœ¨äº†è°ƒä¸åˆ°sklearnçš„åŒ…</p>
<p>*æ‰¾ä¸åˆ°è®­ç»ƒé›†åˆ</p>
<p>-cluster_function_prediction.pyè¦æ”¾åœ¨ç¯å¢ƒé‡Œï¼ˆrightï¼Œç¨‹åºä¼šåœ¨ç¯å¢ƒé‡Œé¢æ‰¾å…¶ä»–æ¨¡å—, æ¯”å¦‚ï¼šcluster_function_prediction_tools.pyï¼‰</p>
<p>*ç¨‹åºè¿è¡Œä»¥åæ‰¾ä¸åˆ°è®­ç»ƒé›†æ•°æ®ï¼Œè§æŠ¥é”™â€œdid not find file containing training data, please keep script located in directory downloaded from githubâ€
*ç°åœ¨ï¼š python3 cluster_function_prediction.py /hdata/tmp/seq/anotherANTISMASH/MK613929/MK613929.gbk /hdata/tmp/seq/anotherRGI/rgi_MK613929.txtï¼Œ ä»»ç„¶è·‘ä¸é€šã€‚ä¸æ˜¯ç¼ºæ¨¡å—ï¼Œæ˜¯ç¨‹åºæ‰¾ä¸åˆ° è®­ç»ƒé›†æ•°æ®
*github (
<a href="https://github.com/allie-walker/Natural-product-function">https://github.com/allie-walker/Natural-product-function</a>)çš„Readmeä»‹ç»
ï¼Œ tranining data åœ¨ç›®å½•RGI_outputä¸‹é¢ã€‚å’Œ antiSMASH_output</p>
<p>*å¦å¤–ï¼Œè¿è¡Œçš„æ—¶å€™ï¼Œè¯´ï¼š ./cluster_function_prediction.py:159: DeprecationWarning: &lsquo;U&rsquo; mode is deprecated
record = SeqIO.read(open(antismash_infilename, &lsquo;rU&rsquo;),&ldquo;genbank&rdquo;)
è¿™ä¸ª â€˜rU&rsquo;  æ¨¡å¼å·²ç»ä¸ç”¨äº†ï¼Ÿæœ‰æ²¡æœ‰æ›¿ä»£çš„ï¼Ÿ</p>
<p>â€˜rU&rsquo;æ”¹æˆ&rsquo;r&rsquo;åº”è¯¥å°±å¯ä»¥äº†</p>
<p>è®¡ç®—ç»“æœæ˜¾ç¤ºï¼Œ
/home/dming/test/npf/Natural-product-function/readInputFiles.py:17: UserWarning: no pfam features found, make sure to run antiSMASH with &ndash;fullhmmer option
warnings.warn(&ldquo;no pfam features found, make sure to run antiSMASH with &ndash;fullhmmer option&rdquo;)
/home/dming/test/npf/Natural-product-function/readInputFiles.py:18: UserWarning: if antiSMASH was run with fullhmmer then this BGC does not have enough similarity to the training set for useful predictions
warnings.warn(&ldquo;if antiSMASH was run with fullhmmer then this BGC does not have enough similarity to the training set for useful predictions&rdquo;)
/home/dming/test/npf/Natural-product-function/readInputFiles.py:20: UserWarning: no resistance features found, double check RGI input file
warnings.warn(&ldquo;no resistance features found, double check RGI input file&rdquo;)</p>
<p>npf æç¤ºï¼ˆè­¦å‘Šï¼‰è¯´ï¼Œæ‰¾ä¸åˆ° è·ŸæŸ¥è¯¢åºåˆ—ç›¸ä¼¼çš„ å·²çŸ¥è›‹ç™½å®¶æ—ã€å’Œè·Ÿå·²çŸ¥æŠ—æ€§åŸºå› åº“ç›¸ä¼¼çš„è‡ªæŠ—åŸºå› ã€‚</p>
<p>é‡è·‘antismash
antismash /hdata/tmp/seq/MK613929.fasta &ndash;genefinding-tool prodigal &ndash;fullhmmer</p>
<p>ylu@hilbert:/hdata/tmp/seq/antismash_MK$ antismash /hdata/tmp/seq/MK613929.fasta &ndash;genefinding-tool prodigal &ndash;fullhmmer</p>
<p>é‡è·‘rgi
cd /hdata/tmp/seq/rgi_MK/</p>
<p>conda activate rgi</p>
<p>rgi main -i /hdata/tmp/seq/MK613929.fasta -o rgi_MK613929</p>
<p>é‡è·‘Natural-product-function
python3 cluster_function_prediction.py /hdata/tmp/seq/antismash_MK/MK613929/MK613929.gbk /hdata/tmp/seq/rgi_MK/rgi_MK613929.txt</p>
<p>è®¤çŸ¥&amp;è¯„ä»·
npf å¯èƒ½æ˜¯ä¸“é—¨é’ˆå¯¹ ç­›æŸ¥ ï¼ˆæ–°ï¼Ÿï¼‰æŠ—ç”Ÿç´  è€Œå¼€å‘çš„ã€‚è¿è¡Œçš„ä¾‹å­ï¼Œè¡¨æ˜å®ƒåŸºæœ¬ä¸Šæ˜¯åœ¨åšè¿™æ ·çš„äº‹æƒ…ï¼š å³åœ¨ç»™å®šçš„åŸºå› ç»„é‡Œé¢ï¼Œæ‰¾å‡ºå¯èƒ½çš„æŠ—æ€§åŸºå› ï¼Œç„¶åç»¼åˆä½¿ç”¨ card/pfam/efi ç­‰å·¥å…·åˆ¤æ–­å…¶äº§ç‰© æ˜¯ æŠ—ç”Ÿç´ ç­‰ æ¦‚ç‡æœ‰å¤šå¤§
çœ‹èµ·æ¥ï¼Œå…¶ç­›æŸ¥ç­‰èƒ½åŠ›åº”è¯¥æ²¡æœ‰è¶…è¿‡ card æ•°æ®åº“åŠ›æ‰€èƒ½åŠçš„èŒƒå›´ã€‚ä¸è¿‡ï¼Œå®ƒå€’æ˜¯ä½¿ç”¨äº†ç°åœ¨å¾ˆå¤š groups ç”¨åˆ°çš„é…¶åŠŸèƒ½åˆ†æçš„å·¥å…· &mdash;-å³EFI å·¥å…·&mdash;è¿™ä¸€ç‚¹è¿˜æ˜¯å€¼å¾—å­¦ä¹ çš„ã€‚æ¢å¥è¯è¯´ï¼Œæˆ‘ä»¬å¯ä»¥åœ¨åç»­çš„coding ä¸­ï¼Œä½¿ç”¨EFIæ¥æé«˜å¯¹åŸºå› åŠŸèƒ½çš„æ³¨é‡Šçš„å‡†ç¡®åº¦å’Œç²¾åº¦
npfæé«˜äº†ä¸€ä¸ªç»¼åˆä½¿ç”¨ cardæ•°æ®åº“ã€ç‰¹åˆ«æ˜¯ EFI åº“çš„æ–¹æ³•ï¼Œå€¼å¾—å€Ÿé‰´ï¼›è¿™ä¸€ç‚¹å¯ä»¥ç”¨åˆ°æˆ‘ä»¬æ‰‹å¤´çš„æŠ—æ€§åŸºå› æ•°æ®åº“æ„å»º/é¢„æµ‹çš„å·¥ä½œä¸­ã€‚</p>
<p>å®˜æ–¹æ•™ç¨‹
Supplemental Computational Methods
antiSMASH runs:
We ran antiSMASH 4.0 on each cluster using the following command:
antismash &ndash;verbose &ndash;outputfolder OUTPUT_FOLDER &ndash;statusfile status.txt &ndash; full-hmmer &ndash;borderpredict INPUT_FILE</p>
<p>RGI runs:
We ran RGI 3.2.1 on each cluster using the following command:
rgi -i INPUT_FASTA_FILE -o OUTPUT_FILE &ndash;loose_criteria YES &ndash;verbose ON &ndash; num_threads 4</p>
<p>Requirements to run prediction script:
We have provided all code necessary to process training set input files and make predictions on novel clusters in a Jupyter notebook. Additionally, we have provided the script to make predictions on novel clusters as a command line application. All code is available here: <a href="https://github.com/allie-walker/Natural-product-function">https://github.com/allie-walker/Natural-product-function</a>. The code was developed to run in Python 3. The libraries required are Biopython, NumPy, scikit-learn, and Matplotlib (for Jupyter notebook code only). Below is a list of the exact versions used to generate data in this paper:
Anaconda 4.9.0
Python 3.8.3
Biopython 1.77
NumPy 1.18.5
Scikit-learn 0.23.1
Matplotlib 3.2.2</p>
<p>Processing antiSMASH output files:
A script for processing antiSMASH output files is available here: <a href="https://github.com/allie">https://github.com/allie</a>walker/Natural-product-function. We used Biopythonâ€™s SeqIO module to read in the genbank output files produced by antiSMASH. We iterated through features in the genbank records and counted the following features: PFAM, CDS motif, smCOG, and polyketide and non-ribosomal peptide monomer predictions. For PFAM features we only included PFAM annotations that had a score of at least 20. All annotations that appeared at least five times in the dataset were included in the final feature list.
Processing RGI output files:
A script for processing RGI output files is available here: <a href="https://github.com/allie-walker/Natural">https://github.com/allie-walker/Natural</a>product-function. We read in the text file produced by RGI and extracted the best hit and corresponding e-value for each best hit. E-values less than or equal to 0.1 were counted as the best hit. All resistant genes that appeared at least five times in the dataset were included in the final feature list.</p>

    
  </div>
</div>

  <div class="pagination__title">
       <span class="pagination__title-h"><span class="valine-comment-count" data-xid="Chocomilk-Leo.github.io/natural-product-function/"></span> æ¡è¯„è®º</span>
       <hr />
  </div>
  <div id="comments"></div>
  <script>
    loadComments();
    function loadComments() {
        if (typeof Valine === 'undefined') {
            var getScript = (options) => {
                var script = document.createElement('script');
                script.defer = true;
                script.crossOrigin = 'anonymous';
                Object.keys(options).forEach((key) => {
                    script[key] = options[key];
                });
                document.body.appendChild(script);
            };
            getScript({
                src: 'https://cdn.jsdelivr.net/npm/valine@1.3.10/dist/Valine.Pure.min.js',
                onload: () => {
                    newValine();
                }
            });
        } else {
            newValine();
        }
    }
    function newValine() {
      new Valine({
        el: '#comments' ,
        recordIP : true,
        appId: '',
        appKey: '',
        placeholder: 'æ—¢ç„¶æ¥äº†ï¼Œé‚£å°±ç•™ä¸ªç—•è¿¹å§~',
        visitor:  true 
      });
      function diyValine() {
        document.querySelector("span.vemoji-btn").innerHTML = 'ğŸ˜€';
        jinrishici.load(function(result) {
            var jrsc_plac =  result.data.content + "\nã€Œ" + result.data.origin.title + "ã€" + result.data.origin.dynasty + " Â· " + result.data.origin.author
            document.getElementById("veditor").setAttribute("placeholder",jrsc_plac);
        });
      }
      setTimeout(diyValine, 1000);
    }
  </script>
  <script src="https://sdk.jinrishici.com/v2/browser/jinrishici.js"></script>
  <script type="text/javascript" src='https://cdn.jsdelivr.net/npm/jquery@3/dist/jquery.min.js'></script>
  <script type="text/javascript">
    jQuery(document).ready(function($){
        $("#comments").on('click', 'span.vat',function(){
        	$(this).parent('div.vmeta').next("div.vcontent").after($("div.vwrap"));
        	$('textarea#veditor').focus();
        })
        if(window.location.hash){
        	var checkExist = setInterval(function() {
        		if ($(window.location.hash).length) {
                $('html, body').animate({scrollTop: $(window.location.hash).offset().top-90}, 1000);
                            clearInterval(checkExist);
                        }
                }, 100);
        }
    });
  </script>

    </main>

    
      
    
  </body>
</html>
